worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Controller.scala

```diff
@@ -324,6 +324,42 @@ private[deploy] class Controller(
       return
     }
 
+    val shuffleCommitTimeout = conf.workerShuffleCommitTimeout
+
+    shuffleCommitInfos.putIfAbsent(shuffleKey, new CommitInfo(null, CommitInfo.COMMIT_NOTSTARTED))
+    val status = shuffleCommitInfos.get(shuffleKey)
+
+    def waitForCommitFinish(): Unit = {
+      val delta = 100
+      var times = 0
+      while (delta * times < shuffleCommitTimeout) {
+        status.synchronized {
+          if (status.status == CommitInfo.COMMIT_FINISHED) {
+            context.reply(status.response)
+            return
+          }
+        }
+        Thread.sleep(delta)
+        times += 1
+      }
```

Comment:
why not use direct response such as commit_in_process to client instead of let client wait network timeout，(otherwise the client may continue to retry due to network exceptions. is it true?)

Reply:
> 

I just added retry logic in client. The design is that worker should always process handleCommitFiles for a particular shuffleKey ONCE. In case (which I think is rare) one handleCommitFiles request comes while another is in process, then the request should wait for timeout. If that happens, the client will trigger requestCommitFiles again if not exceeds maxretries.